README.md

```diff
@@ -62,13 +62,14 @@ npm install -g esy@0.5.7
 - `esy install`
 - `esy '@test' install`
 
-### Building
+### Building from Head
 
 - `esy build`
 
-### Running tests
+### Running tests in a testing environment
 
 - `esy '@test' build`
+- `esy '@test' x run-tests.sh` (This will compile your code first and all tests) 
```

Comment:
This could be reduced to `esy @test && esy test`.

`esy @test` will install and build the test project, a shortcut for `esy @test install && esy @test build`.

`esy test` will run the test script defined in `package.json`, which evaluates to `esy '@test' x run-tests.sh`

Reply:
I got these commands from Bryphe in discord, when we were having problems with `esy test` directly. I'll play around a bit more tonight

Reply:
so the first worked, but the second `esy test` errors on my mac:


```
Lees-MacBook-Pro:src lbot$ esy test
error: project is not installed, run `esy install`
  
esy: exiting due to errors above
```

Reply:
Right, because the script is in the default sandbox you have to have that installed as well. It might be an idea to move/copy that script to test.json as `run`, so you can just do `esy @test && esy @test run`.

Reply:
I'm sorry that I seem so dense, but I tried your suggestion and I'm hitting this silly "project is not installed" error again, and I can't seem to figure it out:

I made this in my test.json:

```
{
  "name": "libvim",
  "version": "8.10869.0",
  "description": "Standalone vim library",
  "license": "MIT",
    "scripts": {"run": "esy x run-tests.sh"},
  "esy": {
    "build": [
        ["bash", "-c", "#{os == 'windows' ? 'build/build-windows-tests.sh' : 'build/build-posix-tests.sh'}"]
    ],
    "buildsInSource": true
  },
  "dependencies": { }
}
```
I then ran `esy '@test'` and got the following output at the bottom of the command:
```

echo "Copying apitest/yank.test.exe to ~/Development/onivim/libvim-test/libvim/src/_esy/test/store/s/libvim-0c7375f1/bin"
Copying apitest/yank.test.exe to ~/Development/onivim/libvim-test/libvim/src/_esy/test/store/s/libvim-0c7375f1/bin
cp apitest/yank.test.exe ~/Development/onivim/libvim-test/libvim/src/_esy/test/store/s/libvim-0c7375f1/bin
cp build/run-tests.sh ~/Development/onivim/libvim-test/libvim/src/_esy/test/store/s/libvim-0c7375f1/bin
```
after, I try to use the new run script, and it get the same error:

```
$ esy '@test' run
error: project is not installed, run `esy install`
  
esy: exiting due to errors above
```

I thought at first that maybe run-tests.sh was doing something weird inside but it's pretty benign. This is basically the same part that was broken before, so I suspect it's something with the way the` esy x run-tests.sh `command is written that esy is not envoking the env properly. What am I missing here? Sorry for the noob questions but esy is eluding me!

Reply:
For what it's worth, I've basically never been able to get anything I put as a script to run (echo $HOME, python --version, etc), I always get the esy project is not installed error.

Reply:
No worries, it really is confusing.

`esy x run-tests.sh` will run `run-tests.sh` in the default sandbox, which means it needs to be installed first. To run it in the `@test` sandbox you'll need to use `esy @test x run-tests.sh`.

Although I'm not entirely sure `esy @test x` is actually needed. You could also try to have the script execute `run-tests.sh` directly.